### PR TITLE
Support read-only mode in tempto

### LINKED ISSUE #105: Tempto should be decoupled from HDFS
HDFS should not be required to run tests using Tempto. This splits to two things:
1. WebHDFS should not be a prerequisite to run Temtpo
2. `tempto-configuration.yaml` configuration should not require any properties in `hdfs` section, `tests.hdfs` should not be mandatory as well.
Currently the simplest test of checking the results of `SELECT * FROM nation` fails when no `hdfs` properties are defined.
E.g.

Prerequisites:
1. Presto installed
2. Hive connector configured
3. [tempto-configuration.yaml](https://github.com/prestodb/tempto/files/193884/tempto-configuration.yaml.txt) (files with `.txt` extension because of GitHub limitations)
4. [test1.sql](https://github.com/prestodb/tempto/files/193886/test1.sql.txt)
5. [test1.result](https://github.com/prestodb/tempto/files/193888/test1.result.txt)
6. [PrestoTests.java](https://github.com/prestodb/tempto/files/193912/PrestoTests.java.txt)

Steps to reproduce:
1. java -jar product-tests.jar

Expected result:
Tests ran (with failure because of not matching output from `test1.result` file)

Actual result:
Bunch of errors complaining about missing configuration (which should not be used):
1) No implementation for java.lang.Integer annotated with @com.google.inject.name.Named(value=hdfs.webhdfs.port) was bound.

2) No implementation for java.lang.String annotated with @com.google.inject.name.Named(value=hdfs.username) was bound.

3) No implementation for java.lang.String annotated with @com.google.inject.name.Named(value=hdfs.username) was bound.

4) No implementation for java.lang.String annotated with @com.google.inject.name.Named(value=hdfs.webhdfs.host) was bound.

5) No implementation for java.lang.String annotated with @com.google.inject.name.Named(value=tests.hdfs.path) was bound.

### LINKED ISSUE #107: TableManager should not be mandatory field in configuration
If user does not need creating any tables `TableManager` should not be mandatory (read-only tests).
Simple scenario to show why current approach is erratic:

Prerequisites:
1. Presto installed
2. Hive connector configured
3. [tempto-configuration.yaml](https://github.com/prestodb/tempto/files/193919/tempto-configuration.yaml.txt) (files with .txt extension because of GitHub limitations) - configuration without table manager specified for Presto (similar to what's specified in docs)
4. [test1.sql](https://github.com/prestodb/tempto/files/193920/test1.sql.txt)
5. [test1.result](https://github.com/prestodb/tempto/files/193923/test1.result.txt)
6. [PrestoTests.java](https://github.com/prestodb/tempto/files/193924/PrestoTests.java.txt)

Expected result:
Test ran (probably failed because of wrong results but at least ran)

Actual result:
7) No implementation for java.util.Map<java.lang.String, com.teradata.tempto.fulfillment.table.TableManager> was bound.

[currently rest of the errors as in https://github.com/prestodb/tempto/issues/105]

### PR DESCRIPTION
**Pull request by kokosing**
Support read-only mode in tempto
    
    By read only I mean a tempto configuration which does not have
    configured hdfs nor a table manager for database connection.
    
    Fixes #107, fixes #105.

### CODE REVIEW

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java [hunk 1]
```diff
@@ -93,7 +86,7 @@ protected void configure()
                                 tableManagerClass -> tableManagerClass));
             }
 
-            @Inject
+            @Inject()
```
Comment (losipiuk) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: why parens?

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java [hunk 2]
```diff
@@ -56,11 +53,7 @@ protected void configure()
 
                 for (String database : databaseNames) {
                     Configuration databaseConfiguration = databasesSectionConfiguration.getSubconfiguration(database);
-                    Optional<String> tableMangerTypeOptional = databaseConfiguration.getString("table_manager_type");
-                    if (!tableMangerTypeOptional.isPresent()) {
-                        continue;
-                    }
-                    String tableManagerType = tableMangerTypeOptional.get();
+                    String tableManagerType = databaseConfiguration.getString("table_manager_type").orElse("read_only");
```
Comment (losipiuk) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: Is there a constant for "read_only" defined somewhere?

File: tempto-examples/src/main/java/com/teradata/tempto/examples/TemptoExamples.java [hunk 3]
```diff
@@ -21,13 +21,12 @@
...
 public class TemptoExamples
 {
-
     public static void main(String[] args)
     {
         TemptoRunnerCommandLineParser parser = TemptoRunnerCommandLineParser
                 .builder("tempto examples")
                 .setTestsPackage("com.teradata.tempto.examples", false)
-                .setConfigFile(DEFAULT_TEST_CONFIGURATION_LOCATION, false)
+                .setConfigFile(DEFAULT_TEST_CONFIGURATION_LOCATION, true)
```
Comment (losipiuk) [inline tempto-examples/src/main/java/com/teradata/tempto/examples/TemptoExamples.java]: unrelated?

File: tempto-core/src/main/java/com/teradata/tempto/internal/hadoop/hdfs/HdfsModuleProvider.java [hunk 4]
```diff
@@ -47,6 +55,15 @@ public Module getModule(Configuration configuration)
             @Override
             protected void configure()
             {
+                Set<String> configurationKeys = configuration.listKeys();
+                if (!configurationKeys.contains(CONF_HDFS_WEBHDFS_HOST_KEY)
+                        || !configurationKeys.contains(CONF_TESTS_HDFS_PATH_KEY)) {
+                    logger.debug("No HDFS support enabled as '{}' or '{}' is configured",
```
Comment (losipiuk) [inline tempto-core/src/main/java/com/teradata/tempto/internal/hadoop/hdfs/HdfsModuleProvider.java]: change to INFO

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java [hunk 5]
```diff
@@ -49,8 +53,13 @@ protected void configure()
                 Configuration databasesSectionConfiguration = configuration.getSubconfiguration("databases");
                 Set<String> databaseNames = databasesSectionConfiguration.listKeyPrefixes(1);
 
-                Map<String, ? extends Class<? extends TableManager>> tableManagerClasses = getTableManagerClassesByType();
+                if (databaseNames.isEmpty()) {
```
Comment (losipiuk) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: why do we need that?

File: tempto-examples/src/main/java/com/teradata/tempto/examples/TemptoExamples.java (hunk 3 above)
Comment (kokosing) [inline tempto-examples/src/main/java/com/teradata/tempto/examples/TemptoExamples.java]: related. I change global configuration to test configuration without table_manager or without databases at all

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java (hunk 2 above)
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: nope, will do

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java (hunk 1 above)
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: I will revert

File: tempto-core/src/main/java/com/teradata/tempto/internal/hadoop/hdfs/HdfsModuleProvider.java (hunk 4 above)
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/hadoop/hdfs/HdfsModuleProvider.java]: I used had that, but then that line was printed multiple times (3 times during startup and about 2 times per test).

File: tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java (hunk 5 above)
Comment (kokosing) [inline tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/TableManagerDispatcherModuleProvider.java]: without that no `newMapBinder` is called and no map is created. Guice fails that map is not created when it tries to access map of query exutoros or table managers
